develop/parachain-devs/runtime-development/FRAME/create-custom-pallet.md

```diff
@@ -0,0 +1,165 @@
+
+    ```rust
+    --8<-- 'code/develop/parachain-devs/runtime-development/FRAME/create-custom-pallet/barebone-pallet-template.rs'
+    ```
+
+    With this scaffold in place, you're ready to start implementing the specific logic and features of your custom pallet. The subsequent sections of this guide will walk you through populating each of these components with the necessary code for your pallet's functionality.
+
+## Pallet Configuration
+
+Every pallet includes a Rust trait called [`Config`](https://paritytech.github.io/polkadot-sdk/master/frame_system/pallet/trait.Config.html){target=\_blank}, which is used to expose configurable options and link your pallet to other parts of the runtime. All types and constants that the pallet depends on must be declared within this trait. These types are defined generically and made concrete when the pallet is instantiated in the `runtime/src/lib.rs` file of your blockchain.
+
+In this step, you'll only configure the common types used by all pallets:
+
+- `RuntimeEvent` - since this pallet emits events, it requires the runtime's event type to handle them. This ensures that events generated by the pallet can be correctly processed and interpreted by the runtime
+- `WeightInfo` - this type defines the weights associated with the pallet's callable functions (also known as dispatchables). Weights help measure the computational cost of executing these functions. However the WeightInfo type will be left unconfigured, since setting up custom weights is outside the scope of this guide
+
+Replace the line containing the [`#[pallet::config]`](https://paritytech.github.io/polkadot-sdk/master/frame_support/pallet_macros/attr.config.html){target=\_blank} macro with the following code block:
+
+```rust
+--8<-- 'code/develop/parachain-devs/runtime-development/FRAME/create-custom-pallet/pallet-config.rs'
+```
+
+## Pallet Events
+
+After configuring the pallet to emit events, the next step is to define the events that can be triggered by functions within the pallet. Events provide a straightforward way to inform external entities such as dApps, chain explorers, or users, that a significant change has occurred in the runtime. In a FRAME pallet, the details of each event and its parameters are included in the node’s metadata, making them accessible to external tools and interfaces.
+
+The [`generate_deposit`](https://paritytech.github.io/polkadot-sdk/master/frame_support/pallet_macros/attr.generate_deposit.html){target=\_blank} macro generates a `deposit_event` function on the Pallet, which converts the pallet’s event type into the [`RuntimeEvent`](https://paritytech.github.io/polkadot-sdk/master/frame_system/pallet/trait.Config.html#associatedtype.RuntimeEvent){target=\_blank} (as specified in the `Config` trait) and deposits it using [`frame_system::Pallet::deposit_event`](https://paritytech.github.io/polkadot-sdk/master/frame_system/pallet/struct.Pallet.html#method.deposit_event){target=\_blank}.
```

Comment:
```suggestion
The [`generate_deposit`](https://paritytech.github.io/polkadot-sdk/master/frame_support/pallet_macros/attr.generate_deposit.html){target=\_blank} macro generates a `deposit_event` function on the `Pallet`, which converts the pallet’s event type into the [`RuntimeEvent`](https://paritytech.github.io/polkadot-sdk/master/frame_system/pallet/trait.Config.html#associatedtype.RuntimeEvent){target=\_blank} (as specified in the `Config` trait) and deposits it using [`frame_system::Pallet::deposit_event`](https://paritytech.github.io/polkadot-sdk/master/frame_system/pallet/struct.Pallet.html#method.deposit_event){target=\_blank}.
```
I'm assuming this is referencing code, if not, please make it lowercase without any backticks